src/components/information-box.ts

```diff
@@ -0,0 +1,57 @@
+import { Control } from 'ol/control';
+import InformationElement from '../types/information-element';
+import { GeocityEvent } from '../utils/geocity-event';
+import SVGCreator from '../utils/svg-creator';
+
+export default class InformationBox extends Control {
+  interval: any;
+
+  constructor(information: InformationElement) {
+
+    const element = document.createElement('div');
+    const title = document.createElement('div');
+    const titleText = document.createElement('div');
+    titleText.innerHTML = information.title
+    const cross = SVGCreator.crossCreator();
+    const content = document.createElement('div');
+    content.innerHTML = information.content
+    const progress = document.createElement('div');
+    element.className = 'custom-popup-element';
+    element.style.setProperty('--progress-width', '100%');
+    title.className = 'custom-popup-title';
+    titleText.className = 'custom-popup-title-text';
+    content.className = 'custom-popup-content';
+    progress.className = 'custom-progress-element'
+    title.appendChild(titleText);
+    title.appendChild(cross);
+    element.appendChild(title);
+    element.appendChild(content);
+    element.appendChild(progress);
```

Comment:
probably easier to use lit for reactivity in those sub components see https://github.com/MediaComem/geocity-wc-map/pull/2

Reply:
Port done